app/routes/task_routes.py

```diff
@@ -0,0 +1,135 @@
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+# ================================
+# Create one task 
+# ================================
+@task_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task.create(request_body)
+    except KeyError:
+        return make_response({"details": "Invalid data"}, 400)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task": new_task.to_json()}), 201
+
+# ==================================
+# Get all tasks  
+# ==================================
+@task_bp.route("", methods=["GET"])
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query == 'asc':
+        all_tasks = Task.query.order_by(Task.title.asc())
+    elif sort_query == 'desc':
+        all_tasks = Task.query.order_by(Task.title.desc())
+    else:
+        all_tasks = Task.query.all()
+    
+    results_list = []
+    for task in all_tasks:
+        results_list.append(task.to_json())
+
+    return jsonify(results_list), 200
+
+# ==================================
+# Get one task by id number
+# ==================================
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    return jsonify({"task": validate_id(Task, task_id).to_json()}), 200
+
+# ==================================
+# Update one task 
+# ==================================
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    request_body = request.get_json()
+    task = validate_id(Task, task_id)
+    task.update(request_body)
+
+    db.session.commit()
+
+    return jsonify({"task": task.to_json()}), 200
+
+# ==================================
+# Delete one task by id
+# ==================================
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = validate_id(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({"details":f'Task {task.task_id} "{task.title}" successfully deleted'}, 200)
+
+# ==================================
+# update one task's completeness
+# ==================================
+@task_bp.route("/<task_id>/<mark>", methods=["PATCH"])
+def mark_tasks_complete_or_incomplete(task_id, mark):
+    task = validate_id(Task, task_id)
+    task.completed_at = datetime.now() if mark == "mark_complete" else None
+    send_to_slack(task.title, "task-notifications", mark)
+    db.session.commit()
+    return jsonify({"task": task.to_json()}), 200
+
+# ==================================
+# Helper function to validate id
+# ==================================
+def validate_id(class_name,id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({"message":f"Id {id} is an invalid id"}, 400))
+
+    query_result = class_name.query.get(id)
+    if not query_result:
+        abort(make_response({"message":f"Id {id} not found"}, 404))
+
+    return query_result
```

Comment:
Two minor things:

1. Since this function is used by both the `goal_routes` and `task_routes` modules, I would recommend pulling it out to a shared `utils` module or something like that. The idea being to signal to a reader where they could find the `validate_id` module that is shared between them both.
2. While your messages here are pretty good, I think they could be even more informative if you used the `class_name` parameter in them. So for example, `f"{class_name} id {id} has an invalid id"` would produce `"Task id foo has an invalid id"`. This would be helpful for the `POST /goals/<goal_id>/tasks` endpoint where you have both `Goal`s and `Task`s you're working with.